core/src/main/java/hudson/model/ItemGroup.java

```diff
@@ -88,4 +89,42 @@ default void onRenamed(T item, String oldName, String newName) throws IOExceptio
      * Internal method. Called by {@link Item}s when they are deleted by users.
      */
     void onDeleted(T item) throws IOException;
+
+    /**
+     * Gets all the {@link Item}s recursively in the {@link ItemGroup} tree
+     * and filter them by the given type.
+     * @since FIXME
+     */
+    default <T extends Item> List<T> getAllItems(Class<T> type) {
```

Comment:
This and others should really be `List<? extends T>` to signal immutability, but that would be source-incompatible for existing callers via `Jenkins`.